runtime/nft_test.go

```diff
@@ -77,7 +77,7 @@ pub contract interface NonFungibleToken {
     // publish for their collection
     pub resource interface CollectionPublic {
         pub fun deposit(token: @NFT)
-        pub fun getIDs(): [UInt64]
+        pure pub fun getIDs(): [UInt64]
```

Comment:
Note that this means the NonFungibleToken will need to be updated to make this function `pure` in order to roll out this change; and therefore that any contracts with resources implementing this interface will need to make their implementations `pure` as well. In most cases this should be a very straightforward change, however.

Reply:
@dsainati1 not better to change post condition ?

Reply:
The whole point of this change is to allow people to call certain functions inside their conditions; so something like the postcondition here:
```
pub fun createEmptyCollection(): @Collection {
        post {
            result.getIDs().length == 0: "The created collection must be empty!"
        }
    }
}
``` 
is a pattern we would like to support.

Reply:
I meant we can change post condition in the standard, to not to use function. Changing standard, then expecting everyone to implement this change (even little) is too complicated. 

I agree supporting pure functions on post/pre is nice and very big improvement. Just I dont like unnecessary breaking change on many contracts if we can avoid.

Reply:
@bluesign How could the condition look like instead?

Reply:
@turbolent maybe `result.ownedNFTs.keys.length` ?

Reply:
No post condition would allow people to create "empty" collections that are not actually empty, which seems rather underhanded and might allow people to sneak malicious objects into others' accounts.

Reply:
@dsainati1 I was thinking like you before ( when I first come here ) actually, but actually I gave up on this. On Flow tendency is contract owner is the **boss**. There is nothing preventing contract owner to sneak objects to your account. 

PS:  I really support your way of thinking.

Reply:
Hmm it looks like the typical (e.g. in TopShot) implementation for `getIDs` is to base it on the `ownedNFT`s array, so I think using `result.ownedNFTs.keys.length` would be similar enough, even if not exactly the same, to work. I would feel comfortable changing it to be that.